pkg/infra/tracing/tracing_test.go

```diff
@@ -5,137 +5,149 @@ import (
-	defer func() {
-		require.NoError(t, os.Unsetenv("JAEGER_AGENT_HOST"))
-	}()
-
-	cfg := setting.NewCfg()
-	ts := &Opentracing{Cfg: cfg}
-	_, err := ts.Cfg.Raw.NewSection("tracing.jaeger")
-	require.NoError(t, err)
-	require.NoError(t, ts.parseSettings())
-	jaegerCfg, err := ts.initJaegerCfg()
-	require.NoError(t, err)
+func TestSplitCustomAttribs_Malformed(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected []attribute.KeyValue
+	}{
+		{input: "key1=value1"},
+		{input: "key1"},
+	}
 
-	assert.False(t, jaegerCfg.Disabled)
-	assert.Equal(t, "example.com:6831", jaegerCfg.Reporter.LocalAgentHostPort)
+	for _, test := range tests {
+		_, err := splitCustomAttribs(test.input)
+		assert.Error(t, err)
+	}
 }
 
-func TestInitJaegerCfg_EnabledViaHostPort(t *testing.T) {
-	require.NoError(t, os.Setenv("JAEGER_AGENT_HOST", "example.com"))
-	require.NoError(t, os.Setenv("JAEGER_AGENT_PORT", "12345"))
-	defer func() {
-		require.NoError(t, os.Unsetenv("JAEGER_AGENT_HOST"))
-		require.NoError(t, os.Unsetenv("JAEGER_AGENT_PORT"))
-	}()
-
-	cfg := setting.NewCfg()
-	ts := &Opentracing{Cfg: cfg}
-	_, err := ts.Cfg.Raw.NewSection("tracing.jaeger")
-	require.NoError(t, err)
-	require.NoError(t, ts.parseSettings())
-	jaegerCfg, err := ts.initJaegerCfg()
-	require.NoError(t, err)
-
-	assert.False(t, jaegerCfg.Disabled)
-	assert.Equal(t, "example.com:12345", jaegerCfg.Reporter.LocalAgentHostPort)
+func TestTracingConfig(t *testing.T) {
+	for _, test := range []struct {
+		Name               string
+		Cfg                string
+		Env                map[string]string
+		ExpectedExporter   string
+		ExpectedAddress    string
+		ExpectedPropagator string
+		ExpectedAttrs      []attribute.KeyValue
+	}{
+		{
+			Name:             "default config uses noop exporter",
+			Cfg:              "",
+			ExpectedExporter: "noop",
+			ExpectedAttrs:    []attribute.KeyValue{},
+		},
+		{
+			Name: "custom attributes are parsed",
+			Cfg: `
+			[tracing.opentelemetry]
+			custom_attributes = key1:value1,key2:value2
+			`,
+			ExpectedExporter: "noop",
+			ExpectedAttrs:    []attribute.KeyValue{attribute.String("key1", "value1"), attribute.String("key2", "value2")},
+		},
+		{
+			Name: "jaeger address is parsed",
+			Cfg: `
+			[tracing.opentelemetry.jaeger]
+			address = jaeger.example.com:6831
+			`,
+			ExpectedExporter: "jaeger",
+			ExpectedAddress:  "jaeger.example.com:6831",
+			ExpectedAttrs:    []attribute.KeyValue{},
+		},
+		{
+			Name: "OTLP address is parsed",
+			Cfg: `
+			[tracing.opentelemetry.otlp]
+			address = otlp.example.com:4317
+			`,
+			ExpectedExporter: "otlp",
+			ExpectedAddress:  "otlp.example.com:4317",
+			ExpectedAttrs:    []attribute.KeyValue{},
+		},
+		{
+			Name: "legacy config format is supproted",
+			Cfg: `
+			[tracing.jaeger]
+			address = jaeger.example.com:6831
+			`,
+			ExpectedExporter: "jaeger",
+			ExpectedAddress:  "jaeger.example.com:6831",
+			ExpectedAttrs:    []attribute.KeyValue{},
+		},
```

Comment:
I would worth having a test if both sections `tracing.jaeger` and `tracing.opentelemetry` exist.